### ISSUE TITLE
[Bug] type为pictorialBar、symbolRepeat为true、containLabel为true 并且没有 symbolBoundingData时，浏览器缩放到极限时cpu 100%且崩溃

### ISSUE THREAD

**Issue by hymmyh** [apache/echarts#20467]
### Version

5.5.0

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?code=PYBwLglsB2AEC8sDeAoFsOzMYAbSIAXMgL4A06mAngIIAeEAzsUuZRnfUy-5nxgBMAhmCHFYAbQCMZWACZZAZlkAWWQFYAuhT5s-AcwBOEAT37nYuAKYAzMMQBEADgCkDnRb7H9AC3uxnNw9PDAAjYDBsAFtHdSDeT2wiAMUABlT3BIsAYxhRCGgAGSFQq1xiMEMAVytePUxGK2MrZkksjFQQzDAqECtHEAhs7GMhXAAhIUNMroxGKijw8oDDK2GZ2fnFvAAlKz6RR0qaja6tpYBZKf0CxylU-NnYc7wAZQgAL37Ye-CQ4VExGkslSIKUqm07VgJF4mhQJCAA&_source=echarts-doc-preview&version=5.5.0

### Steps to Reproduce

1、用下面数据创建echart
```
option = {
    tooltip: {},
    yAxis: {},
    xAxis: {
            data:  [1, 2, 3, 4, 5],
    },
    grid: {
            left: "8%",
            right: "8%",
            bottom: "5%",
            top: "300",
            containLabel: true
    },
    series: [
        {
            type: "pictorialBar",
            symbol: "rect",
            symbolRepeat: "true",
            symbolMargin: "10%",
            symbolSize: 10,
            data: [1, 0, 0, 3, 4],
        }
    ]
}
```

https://echarts.apache.org/examples/zh/editor.html?code=PYBwLglsB2AEC8sDeAoFsOzMYAbSIAXMgL4A06mAngIIAeEAzsUuZRnfUy-5nxgBMAhmCHFYAbQCMZWACZZAZlkAWWQFYAuhT5s-AcwBOEAT37nYuAKYAzMMQBEADgCkDnRb7H9AC3uxnNw9PDAAjYDBsAFtHdSDeT2wiAMUABlT3BIsAYxhRCGgAGSFQq1xiMEMAVytePUxGK2MrZkksjFQQzDAqECtHEAhs7GMhXAAhIUNMroxGKijw8oDDK2GZ2fnFvAAlKz6RR0qaja6tpYBZKf0CxylU-NnYc7wAZQgAL37Ye-CQ4VExGkslSIKUqm07VgJF4mhQJCAA&_source=echarts-doc-preview&version=5.5.0

2、浏览器打开F12
<img width="1744" alt="image" src="https://github.com/user-attachments/assets/696a93cc-bca4-49a4-a1aa-c7e5cd58aa6e">

3、调整浏览器高度到最小
<img width="1677" alt="image" src="https://github.com/user-attachments/assets/6c135190-74b6-48bd-9a81-a8e7d5bc153c">

4、浏览器“性能监控” cpu 100% 然后浏览器崩溃。
<img width="1684" alt="image" src="https://github.com/user-attachments/assets/a8d85f50-b224-4cdb-9982-d6030f5babb2">

### Current Behavior

 type为pictorialBar、symbolRepeat为true、containLabel为true 并且没有 symbolBoundingData时；缩放浏览器出现崩溃问题。

### Expected Behavior

希望在后面的版本中修复这个bug。在缩放浏览器的情况下不要导致cpu 100% ，不要让浏览器崩溃。

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

**Comment by helgasoft** [comment]
yes, was able to reproduce in Chrome and Firefox - just need to resize vertically so that the bars disappear. Then they do not come back and tab freezes. Seems the culprit is _containLabel_, resizing works OK without it.
